### DISCUSSION TITLE
Incentivizing users connecting to Desmos Profile

### DISCUSSION THREAD

**Discussion by kwunyeung** [desmos-labs/desmos#568] [Economy]
I would like to start the discussion on how we can incentivize the users who are willing to link their identities from other platforms to Desmos Profile. I have the following conditions in mind.

1. Linking accounts from chains have higher rating than centralized social accounts
2. Chain accounts with certain bonded tokens are required. This is a proof of contribution
3. Centralized social accounts only valid to be incentivized with enough long history. This is to avoid newly created spam accounts.

This is a way to acquire users and introduce decentralization to centralized social network users. Once the accounts are linked and certain conditions are matched, $DSM will be dropped to the Desmos account. This comes with the following questions to answer.

1. How a user not on Desmos can first have an account and have enough fee to save a `DTag`? Can this be all done in decentralized way automatically?
2. How many $DSM tokens will be in this pool?
3. Should this be another module and the fund is managed by a module account?
4. Can we roll out incentives gradually? For example, we roll out with Twitter first, then Github, LikeCoin, Akash, etc.
5. Can we incentivize in different levels? Higher quality accounts would have more incentives? How we defined the level of quality?
6. Should we disable the transfer of `DTag` when we have this incentive? Will there be any cheating if the `DTag` can be transferred to other accounts?

@RiccardoM @bragaz @dadamu @terencesflam what do you think?

**Comment by rmontagnin** [comment]
I think the best and easiest solution to implement thins kind of incentives would be to simply have an external account that will be responsible to airdropping the tokens. This account can be managed by an application (eg. a Chrome extension) that relies on some APIs to sign and broadcast the transactions. 

The overall schema of how things can work is the following:

![Desmos incentives(1)](https://user-images.githubusercontent.com/9119016/127124428-8531a135-858f-4ed9-97b3-275cb610a4bb.png)

### App login
The user should be able to log into the app using their social network acount (eg. GitHub, Twitter, etc).  
Upon login, the app can perform some validity checks  to make sure the user's accounts are old enough and have enough activities to avoid spam and abuse. 

### Profile creation and linkage
Once the user is authenticated properly, it will now be able to create their Profile and link it to the social network of choice. To do this they will: 
1. Create a Desmos account using the integrated Desmos wallet feature. 
2. The application will perform a new transaction containing a [`MsgGrantAllowance`](https://github.com/cosmos/cosmos-sdk/blob/master/x/feegrant/spec/03_messages.md#msggrantallowance) that enables the user to set the application's account as the fee payer of the transactions. 
3. The user performs a `MsgSaveProfile` and `MsgLinkApplication` transaction, allowing them to create a profile and connect it with the social network of choice.

## How it solves the listed problems
This approach should solve all the above mentioned issues. 


 > 1. How a user not on Desmos can first have an account and have enough fee to save a DTag? Can this be all done in decentralized way automatically?

This will be paid by the application account.

> 2. How many $DSM tokens will be in this pool?

This can be decided at any time, since the account can simply be funded using a `MsgSend` transaction.

> 3. Should this be another module and the fund is managed by a module account?

No, it can simply be a `BaseAccount` managed by the application.
    
> 4. Can we roll out incentives gradually? For example, we roll out with Twitter first, then Github, LikeCoin, Akash, etc.

Sure, this can be done from within the application code, by simply looking at the `MsgLinkApplication` or `MsgLinkChain` data and deciding if the user should get some tokens or not. 

> 5. Can we incentivize in different levels? Higher quality accounts would have more incentives? How we defined the level of quality?

Yes, this can be done from within the application code.

> 6. Should we disable the transfer of DTag when we have this incentive? Will there be any cheating if the DTag can be transferred to other accounts?

No, this is not necessary as the rewards will be airdropped to the Desmos accounts and not the DTags.
